pkg/server/handler/v1alpha1/webhook.go

```diff
@@ -45,6 +46,11 @@ func HandleWebhook(ctx context.Context, tenant, eventType, integration string) (
 	}
 	request := service.HTTPContextFrom(ctx).Request()
 
+	type EventDataWithIntegration struct {
+		*scm.EventData
+		Integration string
+	}
+	var allData = make(map[string][]*EventDataWithIntegration)
 	var data *scm.EventData
 
 	if request.Header.Get(github.EventTypeHeader) != "" {
```

Comment:
Can we use switch case for this four types SCM ?

Reply:
No, the keys in the request header are different for every SCM.
- github.EventTypeHeader
- bitbucket.EventTypeHeader
- svn.EventTypeHeader